File: VB/WindowsApplication1/Component/LookAndFeelSettingsHelper.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.IO
Imports System.Runtime.Serialization.Formatters.Binary
Imports DevExpress.LookAndFeel
Imports System.Windows.Forms

Namespace WindowsApplication1
	<System.ComponentModel.DesignerCategory("")> _
	Public Class LookAndFeelSettingsHelper
		Inherits Component

		Public Sub New()
			RestoreSettings()
			AddHandler Application.ApplicationExit, AddressOf Application_ApplicationExit
		End Sub

		' Fields...
		Private _FileName As String

		<DesignerSerializationVisibility(DesignerSerializationVisibility.Content)> _
		Public Property FileName() As String
			Get
				If String.IsNullOrEmpty(_FileName) Then
					Return "LookAndFeelSettings.save"
				Else
					Return _FileName
				End If
			End Get
			Set(ByVal value As String)
				_FileName = value
			End Set
		End Property


		Private Sub Application_ApplicationExit(ByVal sender As Object, ByVal e As EventArgs)
			SaveSettings()
		End Sub


		Private Sub SaveSettings()
			Save(FileName)
		End Sub

		Private Sub RestoreSettings()
			Load(FileName)
		End Sub
		Public Shared Sub Save(ByVal fileName As String)
			Dim stream As FileStream
			Dim settings As LookAndFeelSettings
			Dim formatter As BinaryFormatter

			settings = New LookAndFeelSettings()
			settings.SkinName = UserLookAndFeel.Default.SkinName
			settings.Style = UserLookAndFeel.Default.Style
			settings.UseWindowsXPTheme = UserLookAndFeel.Default.UseWindowsXPTheme

			stream = New FileStream(fileName, FileMode.Create)
			Using stream
				formatter = New BinaryFormatter()
				formatter.AssemblyFormat = System.Runtime.Serialization.Formatters.FormatterAssemblyStyle.Simple
				formatter.Serialize(stream, settings)
			End Using
		End Sub

		Public Shared Sub Load(ByVal fileName As String)
			If File.Exists(fileName) Then
				Using stream As New FileStream(fileName, FileMode.Open)
					Dim formatter As New BinaryFormatter()
					formatter.AssemblyFormat = System.Runtime.Serialization.Formatters.FormatterAssemblyStyle.Simple
					Dim settings As LookAndFeelSettings = TryCast(formatter.Deserialize(stream), LookAndFeelSettings)
					If settings IsNot Nothing Then
						UserLookAndFeel.Default.UseWindowsXPTheme = settings.UseWindowsXPTheme
						UserLookAndFeel.Default.Style = settings.Style
						UserLookAndFeel.Default.SkinName = settings.SkinName
					End If
				End Using
			End If
		End Sub
	End Class
End Namespace

File: CS/WindowsApplication1/Component/LookAndFeelSettingsHelper.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.IO;
using System.Runtime.Serialization.Formatters.Binary;
using DevExpress.LookAndFeel;
using System.Windows.Forms;

namespace WindowsApplication1
{
    [System.ComponentModel.DesignerCategory("")]

    public class LookAndFeelSettingsHelper : Component
    {

        public LookAndFeelSettingsHelper()
        {
            RestoreSettings();
            Application.ApplicationExit += Application_ApplicationExit;
        }

        // Fields...
        private string _FileName;

        [DesignerSerializationVisibility(DesignerSerializationVisibility.Content)]
        public string FileName
        {
            get { return string.IsNullOrEmpty(_FileName)? "LookAndFeelSettings.save" : _FileName; }
            set
            {
                _FileName = value;
            }
        }
        

        void Application_ApplicationExit(object sender, EventArgs e)
        {
            SaveSettings();
        }


        private void SaveSettings()
        {
            Save(FileName);
        }

        private void RestoreSettings()
        {
            Load(FileName);
        }
        public static void Save(string fileName)
        {
            FileStream stream;
            LookAndFeelSettings settings;
            BinaryFormatter formatter;

            settings = new LookAndFeelSettings();
            settings.SkinName = UserLookAndFeel.Default.SkinName;
            settings.Style = UserLookAndFeel.Default.Style;
            settings.UseWindowsXPTheme = UserLookAndFeel.Default.UseWindowsXPTheme;

            using (stream = new FileStream(fileName, FileMode.Create))
            {
                formatter = new BinaryFormatter();
                formatter.AssemblyFormat = System.Runtime.Serialization.Formatters.FormatterAssemblyStyle.Simple;
                formatter.Serialize(stream, settings);
            }
        }

        public static void Load(string fileName)
        {
            if (File.Exists(fileName))
                using (FileStream stream = new FileStream(fileName, FileMode.Open))
                {
                    BinaryFormatter formatter = new BinaryFormatter();
                    formatter.AssemblyFormat = System.Runtime.Serialization.Formatters.FormatterAssemblyStyle.Simple;
                    LookAndFeelSettings settings = formatter.Deserialize(stream) as LookAndFeelSettings;
                    if (settings != null)
                    {
                        UserLookAndFeel.Default.UseWindowsXPTheme = settings.UseWindowsXPTheme;
                        UserLookAndFeel.Default.Style = settings.Style;
                        UserLookAndFeel.Default.SkinName = settings.SkinName;
                    }
                }
        }
    }
}
